### ISSUE TITLE
Create download page

### ISSUE THREAD

**Issue by gilgongo** [jamulussoftware/jamulussoftware.github.io#322]
This to be the default URL for all platform downloads at https://jamulus.io/downloads/ (or similar URL).

Some examples of different approaches:

https://www.torproject.org/download/
https://www.plex.tv/downloads/
https://ubuntu.com/download/desktop
https://www.videolan.org/vlc/#download

See also https://github.com/jamulussoftware/jamulus/discussions/1023 and https://github.com/jamulussoftware/jamulus/issues/473

**Comment by pljones** [comment]
I'll just say I prefer the Ubuntu one over the VLC one... :)

**Comment by mulyaj** [comment]
Would this use the wiki layout?

**Comment by gilgongo** [comment]
Only if we can't work out anther way :-)

**Comment by mulyaj** [comment]
Made a mockup. Thoughts?

![Jamulus Download Mockup 1](https://user-images.githubusercontent.com/63881221/109870939-aac54580-7c38-11eb-9236-7d1f3c63a963.png)

**Comment by pljones** [comment]
My _only_ reaction: there's a bit too much whitespace vertically - it's more 4x3 than 16x9...  (It may even be taller than wide...)  Looks great other than that!  Should be the jamulus.io front page.

**Comment by gilgongo** [comment]
That works! The problem of course is actually working out the CSS to do it :-) My knowledge of the [framework we're using](http://fox-css.com/documents/) is almost nil I'm afraid.

**Comment by mulyaj** [comment]
@pljones The page includes scroll, so that explains the height, I'll try to reduce the whitespace.

@gilgongo I'll try to figure something out.

**Comment by gilgongo** [comment]
@mulyaj as part of https://github.com/jamulussoftware/jamuluswebsite/issues/192 (which I have in a branch but was waiting until things settled down a bit) I'm putting optional floats on images, but apart from that, I have no real idea how to go about any structural stuff on the site. It would be great if we we could get that top nav you have in that mockup, for example!

**Comment by mulyaj** [comment]
@ann0see Would you have any thoughts on how we would integrate a new page? 

What I was thinking is that it could just be its own standalone HTML and CSS. Would it need a layout and a corresponding .md?

**Comment by gilgongo** [comment]
Well, we're using Jekyll to do the translations (and other bits and pieces). And we'd want the download page to be multi-lingual like the rest of the site. 

But Jekyll is as far as I know independent of Fox CSS, so if we could work out a way of getting the best of both that would good. @ann0see Did the main work for the website though so should know more.

**Comment by ann0see** [comment]
Actually, the current homepage (index.html) is also mostly html + css. It should be quite easy to edit the file + layout. I like your proposal but would add more information about what Jamulus is/... if that becomes our new homepage.

**Comment by ann0see** [comment]
> I'm putting optional floats on images

I'd avoid floats if possible (personal preference).

The framework uses flex which should make it easy to design a layout like that.

**Comment by mulyaj** [comment]
I would probably leave the homepage as is, with the download page as its own thing.

**Comment by ann0see** [comment]
Ok. But what's the point of having a site which has a different layout then? Wouldn't that be confusing?

**Comment by gilgongo** [comment]
@ann0see the floats are only on the images, so that we can wrap text around them. But see below...

@mulyaj  good point! If we can come up with some page templates and consistent top nav that would be ideal. I'll have a think on some based on what we have content-wise. Dunno what the overall possibilities are though.

**Comment by mulyaj** [comment]
Well the value of the homepage is that we can have content that gives a broad overview of what Jamulus is, and does (which I think it does a decent job at).

I'm not too fond of presenting a user with a huge download button right from the get-go.

**Comment by ann0see** [comment]
> and consistent top nav that would be ideal.

Should be possible. But you'll need to think about mobile devices too. What would they show in the top nav?

**Comment by mulyaj** [comment]
>What would they show in the top nav?

I think for mobile devices, we can leave the nav button as is - and have desktop screens show the various page links on the right (downloads, wiki, getting started, etc).

**Comment by gilgongo** [comment]
BTW the term "wiki" here is pretty confusing, so best not surface it in public :-)

**Comment by pljones** [comment]
So it doesn't get lost (in my memory and general stream of consciousness)...

The landing page for the download should both tell a user where they are - "This is Jamulus: Play music online. With friends. For free." and say where to get it (i.e. just add the banner with the catchphrase to the initial idea and trim the vertical spacing...).

**Comment by mulyaj** [comment]
@pljones something like this? (the tagline is a bit blurry in this picture, but otherwise it's clear)

edit: I'm playing around with using a different font (Arial seems very "default" to me). Here I'm using Lato.

![Jamulus Download Mockup 2](https://user-images.githubusercontent.com/63881221/110156761-5bf0ea80-7db5-11eb-90d2-195b490881a6.png)

**Comment by gilgongo** [comment]
# Looks good! 

**Random thought 1**

On the breadcrumb, the ones I put in on the changes branch were very "conservative" in the sense that they would only show at the second level downwards. So for example, the [Windows installation page](https://jamulus.io/wiki/Installation-for-Windows), which is a child of Getting Started, which is a child of Using Jamulus:

```
Using Jamulus
   Getting Started
      Windows
```
So that page has:

Using Jamulus > [Getting Started](https://jamulus.io/wiki/Getting-Started)

as its breadcrumb (the "leaf" being the page's H1 - not repeated by the breadcrumb).

If we want to have breadcrumbs on _every_ page other than the home page, then that's of course possible. I just thought it would be a bit redundant on most pages. For example in the screen you have above, the link to "Downloads" would do nothing because that's the page you're on :-) And "home" is by convention the top left logo.

There's possibly an advantage in repeating the H1 of the page as the leaf of the breadcrumb if they have "mutually supporting" names (so the H1 might be "Installation for Windows" while the leaf might be "Windows users" or something). But that's quite tricky in some cases. 

**Random thought 2**

Centred text ("Got a pair of headphones...") can be confusing to read for some people with neurodiversity (dyslexia mainly I think). 

**Random thought 3**

Would be good to get a footer that works across the site a bit better too...

**Random thought 4**

Not sure how the "Download for OS" is going to work in the short term. We don't have an installer yet for Linux, so the "compile from source" link is a little funky (and assuming the number of people who would want to compile it for another platform are going to just go to GitHub anyway).

**Comment by mulyaj** [comment]
> On the breadcrumb, the ones I put in on the changes branch were very "conservative" in the sense that they would only show at the second level downwards. 

Hm, I see. I'm curious if there really is a point in adding a breadcrumb? It seems like the side-nav already serves the purpose of locating the user. 

> Not sure how the "Download for OS" 

For this button, I was thinking it would have a dropdown similar to the one on [VLC](https://www.videolan.org/vlc/#download), to let you select your OS. For Linux, maybe that could be omitted from the main download button and be given a separate blurb? 

The "compile from source" link is supposed to go to [this instruction page](https://jamulus.io/wiki/Compiling), which is where our Github Readme links as well.  Maybe I could change the copy to make that clearer.

**Comment by gilgongo** [comment]
> I'm curious if there really is a point in adding a breadcrumb?

When things are in the nav, it's signalled, yes (although weakly, there's not a strong visual convention for navs). It just means we can handle "orphans" like [this](https://jamulus.io/wiki/Hardware-Setup) and [this](https://jamulus.io/wiki/Running-a-Private-Server), which aren't on the nav. An alternative being to put them there, and have a rather large nav as a consequence :-) Some sites have "site maps" in their footers for this reason, which sometimes work quite well though.  

But generally, a breadcrumb is just a reasonably well-recognised convention to "orientate" you when things get non-obvious as to what they relate to. There is also an SEO consideration when people find these pages on Google too.

> For this button, I was thinking it would have a dropdown

Ah yes that would be nice.

> The "compile from source" link is supposed to go to this instruction page

I see. As compiling only affects maybe less than 0.25% of the audience, it's not a big deal as long as "non-compiling" people (who BTW may not even know what "compiling" means) don't get confused by it.

**Comment by mulyaj** [comment]
>"orphans" like this and this, which aren't on the nav.

Right, I totally forgot about those pages. Off-topic, but for some "orphans" they're really only accessible on certain pages Plus, with the fact that they're mostly inline links, they can be easily missed. We can probably discuss this elsewhere.

Anyways, I won't include a breadcrumb for now.

**Comment by gilgongo** [comment]
> that they're mostly inline links, they can be easily missed.

Yes, this is sort of deliberate in some cases. If we put pages [like this](https://jamulus.io/wiki/Central-Servers) on the nav it would make Jamulus look too daunting for new users if they thought they would need to read all that stuff. We also have them in line so that we can make sure, for example, you read [this](https://jamulus.io/wiki/Running-a-Server) before you read [this](https://jamulus.io/wiki/Running-a-Private-Server). That "iceberg" structure is in fact pretty common on a lot of sites (but hard to spot of course because it's not on the nav!).

**Comment by pljones** [comment]
I think having an "Ubuntu Linux (Full)" link and an "More..." link in the drop down would be better than mentioning "Compile from source".

I think Ubuntu's probably common enough and we provide an installer - double-click on the downloaded .deb should work.

"Other" would take you to the place where everything got explained in full - it should be only a small number of people who'd be expected to be those who know what they're doing who go there.  It would provide the remaining builds (Ubuntu headless, Android) and either direct to the compile page or be the compile page.

**Comment by mulyaj** [comment]
Ok, here's a preview of the page with html/css - I decided not to use the dropdown as I don't think it's entirely necessary for just 3 options.

![image](https://user-images.githubusercontent.com/63881221/111043023-321b7180-840e-11eb-8603-eb0a534a967c.png)
![image](https://user-images.githubusercontent.com/63881221/111043034-4bbcb900-840e-11eb-890f-6df34b8f5565.png)
<img src="https://user-images.githubusercontent.com/63881221/111043159-0cdb3300-840f-11eb-971e-f74a1fb47057.png" width="200" height="750">

**Comment by gilgongo** [comment]
Yes, that's better than a drop-down (which of course "hides" the options). The convention is to use the OS icons to draw the eye, but it's not big deal. 

Super-tiny thing: I'd say "Download" (as in the verb) not "Downloads" as that former is more common.

Otherwise, looks good to me! We do we need to resolve where exactly all the links are going but maybe do that later?

Also BTW how does the nav look on on the home page?

**Comment by mulyaj** [comment]
Here's how the nav looks on the home page: 
edit: the double navbar is a screenshot bug
![image](https://user-images.githubusercontent.com/63881221/111074477-80d21580-84b9-11eb-9461-a6efa484b08e.png)

**Comment by mulyaj** [comment]
>We do we need to resolve where exactly all the links are going but maybe do that later?

Yeah, I think [as you mentioned here](https://github.com/jamulussoftware/jamuluswebsite/issues/349#issuecomment-798033204) the nav links will need their own hub pages - maybe start a new issue once this is merged?

The icons that I used also need to be attributed to the author - do they have to be attributed on the same page they're used? If we make have a team page, I was thinking we could put it there.

After that, looks like we'll need a Linux download and a changelog page.

**Comment by gilgongo** [comment]
Attributions can go anywhere logical really, so yes maybe on the Team page (seeing as we're also thinking of creating that too now https://github.com/jamulussoftware/jamuluswebsite/issues/355) or even just in the main repo readme.md would be OK (there is a general licence credits section already).

EDIT: Home page looks much better now too!

**Comment by pljones** [comment]
Is "Linux" appropriate to every distribution?  The deb files will only be of interest to Debian/Ubuntu/derivative distributions.

**Comment by mulyaj** [comment]
>Is "Linux" appropriate to every distribution?

I'm not too familiar with Linux - would it be "Download for Ubuntu Linux"?

**Comment by gilgongo** [comment]
I think we imply that we would either have to have **another** hub page for the (less than 5% of our) audience that use Linux, so that we can tell them about .debs, compling, PPAs, repos, flatpaks and god knows what else might come along. Or just take them to [this page](https://jamulus.io/wiki/Installation-for-Linux) for now. It's less than 5% after all - and the page has been updated to be _much_ more complex now to keep everyone happy :-)

**Comment by pljones** [comment]
Unfortunately, that's out of date, because we _do_ provide a better mechanism than installing from source for Debian/Ubuntu distributions -- the .deb files.  I'd rather not route to anything suggesting that the best way is from source, if there's an alternative.  The .deb files should get most users to the same point that the Windows installer does.

**Comment by gilgongo** [comment]
Fine by me. So if we assume it's .deb for the download page then that makes it all much simpler!  And one day, we can relegate the compilation instructions to a repo .md file where they rightfully belong (along with [this page](https://jamulus.io/wiki/Compiling) which has always irked me)

**Comment by pljones** [comment]
> Here's how the nav looks on the home page:

...
I'd make the "Downloads" link on the top bar highlighted some way -- I struggled to find it.  There's a _lot_ of text on that page to work through if all you're looking for is the downloads.

**Comment by gilgongo** [comment]
> highlighted some way

It's the Main Nav :-) If that's not prominent enough for you, then nothing will be (also don't forget that unless your'e running at something bigger than 4K UHD you probably won't see half of that text in the screenshot).

**Comment by pljones** [comment]
It doesn't look like "Nav" very much.  There's nothing indicating the words on the right are click-able links to anywhere.  Button-ish styling might help?

**Comment by mulyaj** [comment]
I think the purpose of the top nav is clear - it's a fairly common practice to layout a nav like that.  Don't forget that the links will have hover stylings too.

Edit: I could add a border styling to the download button

As @gilgongo mentioned as well, the navbar is a lot more prominent when actually viewed on a screen.

I could run a few usability tests to verify if necessary.

**Comment by gilgongo** [comment]
> Button-ish styling might help?

While it's not what you mean exactly, there should be a convention of "you are here" styling, so the top nav item for "Download" highlights in some way when you're on the download page". It's actually required for assistive tech think?

EDIT: It's there on the other shots I see.

Also BTW we can always take some of that verbiage on the main page out. It was pretty much the first draft I came up with (subsequently tweaked and expanded somewhat by others):

_"Jamulus is for playing, rehearsing, or just jamming with your friends, your band or just anyone you find online. With high quality, low-latency sound on a normal broadband connection, it's easy to play together remotely and in time. Have a look at how people do it - and [download it here]!"_

(this is forking the discussion I realise, sorry)

**Comment by pljones** [comment]
> (this is forking the discussion I realise, sorry)

Well, getting to the download page makes creating one worthwhile :)

**Comment by mulyaj** [comment]
Added some stylings to the nav, have a look.

I'm not entirely sure how to best implement the "you are here" highlighted link (my JS isn't too good, I could hardcode it in the css but I'm not sure if that's a good idea) 

https://streamable.com/vqr0m1

**Comment by pljones** [comment]
That also makes the top nav more noticeable in general, which is good.  Just enough to draw attention without looking too pushy :).

**Comment by gilgongo** [comment]
Closing this as it now pretty much duplicates https://github.com/jamulussoftware/jamuluswebsite/issues/349 and is on the same PR anyway.

**Comment by ann0see** [comment]
If any more work is planned, I think an new issue should be opened. @mulyaj @Riley-I should be aware of this.
